state/event_table_test.go

```diff
@@ -1165,52 +1166,59 @@ func TestEventTable_SelectLatestEventsBetween_MissingPrevious(t *testing.T) {
 	}
 
 	testcases := []struct {
+		Desc            string
 		FromIDExclusive string
 		ToIDInclusive   string
-		Limit           int
 		ExpectIDs       []string
 	}{
 		{
+			Desc:            "has no gaps - should omit nothing",
 			FromIDExclusive: "start",
 			ToIDInclusive:   "C",
 			ExpectIDs:       []string{"C", "B", "A"},
```

Comment:
I find it weird that we reverse the ordering here, but whatever.

Reply:
I was also surprised!

Reply:
AFAICS the only call site of SelectLatestEventsBetween is

https://github.com/matrix-org/sliding-sync/blob/d3ba1f1c30017a1dbaaa654bbae8ba2e9c12661b/state/storage.go#L674-L689

so it won't be too hard to fix this up in the future, if we want to.